lib/entitlement_info_wrapper.dart

```diff
@@ -61,8 +61,8 @@ class EntitlementInfo {
       this.billingIssueDetectedAt);
 
   factory EntitlementInfo.fromJson(Map<dynamic, dynamic> json) {
-    var periodType;
-    switch (json["periodType"] as String) {
+    late var periodType;
+    switch (json["periodType"] as String?) {
```

Comment:
what does this switch do if periodType is null? do we need to add a case to handle it?

Reply:
If `periodType` is `null` then no case will match and periodType will still NOT be initialized.

And when it‘s then passed to `EntitlementInfo()` constructor as it‘s expected to be non-null it will crash.

There are three options:
- Leave it as it is and let it crash.
- Add a `case null: ...` to handle the null case.
- Make `periodType` nullable `String?` and be explicit that it may be null and handle it apprioriately.

I personally would prefer the last. So doing this:

```dart
String? periodType;
```

And the later:
```
EntitlementInfo(
  ...,
  periodType!,
  ...
  // Maybe this for the store (?!)
  store ?? Store.unknownStore,
);
```

I think you are better suited to judge what consequences would be acceptable.
Completely, preventing any errors while parsing is probably never possible.